app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1380,12 +1383,14 @@ impl GraphEditorModelWithNetwork {
 
 // === Node Creation ===
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Debug)]
```

Comment:
What is the reason to make this struct so heavyweight? Why do you need to keep `EdgeEndpoint` instead of `NodeId`, just like `DroppingEdge` does?

Reply:
It was by my request. If we want to support multiple output ports (and we want eventually), then the future implementor would already know where to start.

Reply:
I am OK with adding the information. My question was why instead of adding simply `NodeId` we are adding the whole structure there.